### PR TITLE
Content restructuring

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/14783

## Staging site

You can preview these changes on [this staging site](https://docs-staging.c68pn.k8s.gollum.westeurope.azure.gigantic.io/).

**Caveats:** Search is not working on staging.

## List of changes in this PR

- Menu changed from flat, horizontal to hierarchical, vertical
- All content moved into new structure
- Front page changed
- Navigation pages added
- Some articles have been split up to create "Overview" articles.
- Obsolete front matter fields removed (tags, categories, type, layout)
- Breadcrumb logic and position changed
- Internal links changed to use the `relref` shortcode. As this is validated on HUGO build, the internal link check in the CI config is removed.
- Twitter Hot URLs example is being removed (fixes https://github.com/giantswarm/giantswarm/issues/14143)
- CRD page template used by crd-docs-generator is now part of the docs repo, allowing to make template modifications without the need to touch and release giantswarm/crd-docs-generator.
- Add `make lint-chart` to allow validating the helm chart
- Adapted the TOC (table of content) config in HUGO, which allows to remove some Javascript and CSS code for hiding parts.

### Fixes TODO

- [x] At viewport width ~1000px, the "On this page" TOC gets pushed to the bottom.

### TODO before merge

- [x] Revert all occurrences of `docs-staging-app`
- [x] Remove ingress manifest
- [ ] Use comment `do not release` to prevent automatic release

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -131,7 +136,7 @@ Special front matter fields we use:
 In order to link to other docs pages, please use this format only:
 
 ```markdown
-... see the [gsctl reference](/reference/gsctl/) for details. ...
+... see the [gsctl reference]({{< relref "/ui-api/gsctl" >}}) for details. ...
```
Comment (ubergesundheit) [inline CONTRIBUTING.md:139]: This page is not rendered by hugo, right? The `relref` shortcode does not work on GitHub

**Comment by stone-z** [comment]
Low prio/not sure how much control over this hugo allows, but on vertical screens or thin windows, the "On this page" nav normally on the right side is forced to the bottom: 
image

File: src/content/getting-started/_index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+linkTitle: Getting started
+title: Getting started with Giant Swarm
+description: Guides that help you going with your new Giant Swarm installation, quickly. Things most of our users want to know on day one.
```
Comment (stone-z) [inline src/content/getting-started/_index.md:4]:
```suggestion
description: Guides to help you get going with your new Giant Swarm installation, quickly. Things most of our users want to know on day one.
```

File: src/content/general/architcture/aws/index.md [hunk 3]
```diff
@@ -1,9 +1,11 @@
 ---
+linkTitle: AWS
 title: The Giant Swarm AWS platform
-description: Description of the Giant Swarm platform on Amazon Web Services (AWS), how it looks like and which features offers
-weight: 50
-type: page
-categories: ["basics"]
+description: Description of the Giant Swarm platform on Amazon Web Services (AWS), how it looks like and which features it offers.
```
Comment (stone-z) [inline src/content/general/architcture/aws/index.md:4]: Recommend dropping the "like" and adding Oxford comma. Same for Azure
```suggestion
description: Description of the Giant Swarm platform on Amazon Web Services (AWS), how it looks, and which features it offers.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (marians) [reply CONTRIBUTING.md:139]: Right. Must have been a mass edit failure. 👍

Comment (marians) [reply CONTRIBUTING.md:139]: Wait, that's only a `relref` example for contributors. The code is not supposed to work.

Comment (ubergesundheit) [reply CONTRIBUTING.md:139]: ah yes. the ```markdown line :)

File: Makefile [hunk 4]
```diff
@@ -3,6 +3,9 @@ COMPANY=giantswarm
 REGISTRY=quay.io
 SHELL=bash
 MARKDOWNLINT_IMAGE=06kellyjac/markdownlint-cli:0.23.0-alpine
```
Comment (stone-z) [inline Makefile:5]: Could we retag this?

**Review by stone-z — APPROVED** [review]
Overall this feels much more organized to me. In addition to my comment about narrow windows, I left some content touch-ups I happened to notice, but those can be done separately.

Two other comments:
- the left nav and breadcrumb font might be a little faint. Depending on the screen I put it on and the angle, it can be a little straining to read. Maybe a slightly darker or slightly more bold font would be better. It also might just be me and I'm totally bonkers. Maybe others have thoughts.
- seeing it live, I wonder if it wouldn't make sense to move the "Advanced Setup" further down, below App Platform?

File: Makefile (hunk 4 above)
Comment (marians) [reply Makefile:5]: I made a note in the SIG Front End board

**Comment by marians** [comment]
> Low prio/not sure how much control over this hugo allows, but on vertical screens or thin windows, the "On this page" nav normally on the right side is forced to the bottom

This is fixed with the last commit and will be rolled out to the staging site soon.

File: .circleci/config.yml [hunk 5]
```diff
@@ -68,15 +54,18 @@ workflows:
       - build:
           filters:  # required since `architect/push-to-app-catalog` has tag filters AND requires `build`
             # Ignore all commits, but run when tagging a release.
-            branches:
-              ignore: /.*/
+            
+            # TODO: uncomment the two lines below again after docs-staging-app is no longer a thing
+            #branches:
+            #  ignore: /.*/
```
Comment (marians) [inline .circleci/config.yml:60]: To be reverted before merging

File: .circleci/config.yml [hunk 6]
```diff
@@ -68,15 +54,18 @@ workflows:
...
             tags:
               only: /^v.*/
       - architect/push-to-app-catalog:
           name: "package and push docs"
           app_catalog: "giantswarm-operations-platform-catalog"
           app_catalog_test: "giantswarm-operations-platform-test-catalog"
-          chart: "docs-app"
+          # TODO: revert docs-staging-app to docs-app
+          chart: docs-staging-app
```
Comment (marians) [inline .circleci/config.yml:68]: To be reverted before merging

File: Makefile [hunk 7]
```diff
@@ -73,7 +76,7 @@ clean:
 	rm -rf .sass-cache
 
 # Verify internal links
-linkcheck-external:
+linkcheck-internal:
```
Comment (marians) [inline Makefile:79]: Boyscouting

File: helm/docs-staging-app/Chart.yaml [hunk 8]
```diff
@@ -1,5 +1,6 @@
 apiVersion: v1
-name: docs-app
+# TODO: revert from docs-staging-app to docs-app
+name: docs-staging-app
```
Comment (marians) [inline helm/docs-staging-app/Chart.yaml:3]: To be reverted before merging

File: helm/docs-staging-app/templates/ingress.yaml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+            pathType: Prefix
+            backend:
+              service:
+                name: {{ .Values.name }}
+                port:
+                  number: 8080
+  tls:
+    - hosts:
+      - docs-staging.c68pn.k8s.gollum.westeurope.azure.gigantic.io
+      secretName: {{ .Values.name }}
```
Comment (marians) [inline helm/docs-staging-app/templates/ingress.yaml:29]: To be reverted before merging

File: helm/docs-staging-app/values.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# TODO: revert from docs-staging-app to docs-app
+name: docs-staging-app
+# TODO: revert from docs-staging to docs
+namespace: docs-staging
```
Comment (marians) [inline helm/docs-staging-app/values.yaml:4]: To be reverted before merging

**Comment by stone-z** [comment]
Is Heapster still relevant? https://docs-staging.c68pn.k8s.gollum.westeurope.azure.gigantic.io/getting-started/metrics/#heapster

**Review by puja108 — APPROVED** [review]
LGTM, I guess everyone's feedback and comments will be a good follow-up todo list of nitpicks and minor improvements, but this is definitely already a few steps up from where we're coming from, so rather merge quickly and then improve from there

**Comment by marians** [comment]
> Is Heapster still relevant? https://docs-staging.c68pn.k8s.gollum.westeurope.azure.gigantic.io/getting-started/metrics/#heapster

I don't know. Let's ask @giantswarm/sig-customer-happiness who own that page.

**Review by T-Kukawka — APPROVED** [review]

**Pull request merged**